src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -892,5 +954,58 @@ public int hashCode() {
     public String toString() {
         return "transaction " + hash.toString();
     }
-}
 
+    /**
+     * Puts the TVM in cache
+     * @param tangle Tangle
+     * @param transactionViewModel The tvm to cache
+     * @param hash the hash of the tvm
+     * @throws Exception Exception
+     */
+    private static void cachePut(Tangle tangle, TransactionViewModel transactionViewModel, Hash hash) throws Exception {
+        Cache<Indexable, TransactionViewModel> cache = tangle.getCache(TransactionViewModel.class);
+        if (cache.getSize() == cache.getConfiguration().getMaxSize()) {
+            cacheEvict(tangle);
+        }
+        cache.put(hash, transactionViewModel);
+    }
+
+    /**
+     * Evict {@link CacheConfiguration#getEvictionCount()} items from the cache
+     * @param tangle Tangle
+     * @throws Exception Exception
+     */
+    public static void cacheEvict(Tangle tangle) throws Exception {
+        Cache<Indexable, TransactionViewModel> cache = tangle.getCache(TransactionViewModel.class);
+        for (int i = 0; i < cache.getConfiguration().getEvictionCount(); i++) {
+            Indexable hash = cache.nextEvictionKey();
+            if(hash != null){
```

Comment:
```suggestion
            if (hash != null) {
```
Since you fixed formatting a lot